java_src/foam/dao/SequenceNumberDAO.java

```diff
@@ -0,0 +1,62 @@
+package foam.dao;
+
+import foam.core.ClassInfo;
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+import foam.mlang.sink.Max;
+
+public class SequenceNumberDAO
+  extends ProxyDAO
+{
+  protected String property = "id";
+  protected int value = 1;
+  private PropertyInfo property_ = null;
+  
+  public AbstractDAO setOf(ClassInfo of) {
+    AbstractDAO ret = super.setOf(of);
+    this.property_ = null;
+    return ret;
+  }
+  
+  public SequenceNumberDAO setProperty(String property) {
+    this.property = property;
+    this.property_ = null;
+    return this;
+  }
+  
+  public SequenceNumberDAO setValue(int value) {
+    this.value = value;
+    return this;
+  }
+  
+  private PropertyInfo getProperty_() {
+    if (property_ != null) {
```

Comment:
Simpler to do:

  if ( property_ == null ) {
    property_ = (PropertyInfo) this.of_.getAxiomByName(property);
  }
     
  return property_;